### DISCUSSION TITLE
Mel spectrogram computation and audio pre-processing

### DISCUSSION THREAD

**Discussion by napulen** [openai/whisper#269] [Q&A] [answered]
A quick question on stuff seen in the `audio.py` module.

https://github.com/openai/whisper/blob/9e653bd0ea0f1e9493cb4939733e9de249493cfb/whisper/audio.py#L121-L122

Is `L122` a form of clipping like the one of `L121`? Couldn't you use `torch.clamp` for both of these? Just curious

https://github.com/openai/whisper/blob/9e653bd0ea0f1e9493cb4939733e9de249493cfb/whisper/audio.py#L123

Is that some form of normalization/binding the output to a range? 

I'm also curious about the constants `8.0` and `4.0`. I'm assuming these are just reasonable approximations, would you expect the performance to change significantly if these values were changed?

Any hints would be appreciated. Thank you!

**Comment by jongwook** [comment, marked as answer]
The first set of operations are written to be equivalent to what's done in [librosa's `amplitude_to_db`](https://librosa.org/doc/0.9.2/generated/librosa.amplitude_to_db.html#librosa.amplitude_to_db), which uses the default `top_db` value of 80.0, which sets how small a value in the spectrogram can be compared to the largest value.

After those two lines, the values of `log_sepc` are roughly (but not strictly) in [-8.0, 0.0], and L123 puts them in [-1.0, 1.0] which is typical as an input to deep learning models.

A model trained from scratch without L123 would likely just work as well, but you may see significantly degraded performance without those lines on the released Whisper models because they expect the inputs in that range, and without L123 the input becomes out-of-distribution.

**Reply by napulen** [reply]
That all makes sense. Thank you very much!
